### PR TITLE
FEATURE: Add new/unread counts to tags section links exp sidebar

### PR DESCRIPTION
**Pull request by tgxworld**
This allows us to accurately count the unread/new in Sidebar for each
tag.

### CODE REVIEW

**Comment by tgxworld** [comment]
@OsamaSayegh Can you review this for me? Thank you :bow:

File: app/assets/javascripts/discourse/app/components/sidebar/tags-section.js [hunk 1]
```diff
@@ -7,7 +7,16 @@ export default class SidebarTagsSection extends GlimmerComponent {
...
+      return new TagSectionLink({
+        tag: trackedTag,
+        topicTrackingState: this.topicTrackingState,
+      });
+    });
+  }
+
+  willDestroy() {
+    this.sectionLinks.forEach((sectionLink) => {
+      sectionLink.teardown();
```
Comment (OsamaSayegh) [inline app/assets/javascripts/discourse/app/components/sidebar/tags-section.js:19]: Hmm if `this.currentUser.trackedTags` changes and `sectionLinks` is re-computed, is there anything that calls `teardown()` on the array elements of `sectionLinks` before it's re-computed?

(we may have the same problem with `categories-section`, it didn't occur to me when I reviewed the PR yesterday)

Comment (tgxworld) [reply app/assets/javascripts/discourse/app/components/sidebar/tags-section.js]: @OsamaSayegh Good catch! I've fixed this PR and will follow up with another PR for the other sections.

File: app/assets/javascripts/discourse/app/components/sidebar/tags-section.js [hunk 2]
```diff
@@ -4,10 +4,27 @@ import GlimmerComponent from "discourse/components/glimmer";
...
 export default class SidebarTagsSection extends GlimmerComponent {
+  constructor() {
+    super(...arguments);
+
+    this.callbackId = this.topicTrackingState.onStateChange(() => {
+      this.sectionLinks.forEach((sectionLink) => {
+        sectionLink.refreshCounts();
+      });
+    });
+  }
```
Comment (OsamaSayegh) [inline app/assets/javascripts/discourse/app/components/sidebar/tags-section.js:15]: Awesome, this is a nice solution!

**Review by OsamaSayegh — APPROVED** [review]

**Comment by tgxworld** [comment]
Thank you @OsamaSayegh for reviewing :+1:

**Pull request merged**